### DISCUSSION TITLE
How to deal with empty pages ?

### DISCUSSION THREAD

**Discussion by Limezy** [athensresearch/athens#1026] [Question / Answer]
I don't have any use of the daily page on my Athens workflow.
That's not a problem, although I would be pleased if I could change my startup behaviour (see https://github.com/athensresearch/athens/discussions/1021)
The problem is that every day I'm using Athens, that will create an empty page that I don't need, which will in the long term clutter my page list and my graph !

Thinking about this with members in the discord community striked new thoughts : 
### **How shall we deal with empty pages ? And what *is* actually an empty page ?**

Let me put here my convictions on this, although this is obviously only IMHO and to be discussed below : 
1. We should call "orphan" a page that has no link to it or in it (but that may have many blocks inside)
2. We should call "void" a page that has no blocks in it (but that may have many links to it)
3. We should call "empty" a page that has no blocks inside, and no links to it, that is a page which is both orphan AND void
4. We should be able to hide void pages and/or empty pages from the page list
3. We should be able to hide orphan pages and/or empty pages from the page list

Now back on daily pages, a few thoughts : 
* A daily page is not exactly a normal page, because of the look of it (like a 3D card)
* Apart from this, a freshly created daily page is just an empty page as per above definition, so I should be able to filter it out
* However, some users may want to be able to filter daily pages from the graph, as it's done in Logseq for instance

In the end, all of these could in the end be part of a filtering system for Page list and graph view, so I'm not sure if Athens team would want to implement these features as standalone or just to add them as options in their (hopefully) coming graph and filter pages

**Comment by Limezy** [comment]
Additionally, especially in a few months or years when we will all have our Athens graphs completely clustered, we may want to have special pruning functions, like "remove all empty pages".
But again, this could only be part of a well made filtering system, where one would go in the page list, filter by empty page, select all, delete and that's it.

**Reply by jefftangx** [reply]
This comment has more to do with Daily Notes than empty pages in general.

Juho, an early design contributor, had really interesting designs for [Daily Notes scrubbing](https://www.figma.com/file/iCXP6z7H5IAQ6xyFr5AbZ7/Product-Design-Sandbox?node-id=765%3A6243)

I know @shanberg  had some thought there too. 
![image](https://user-images.githubusercontent.com/8952138/115979339-3b821880-a53a-11eb-9d45-2f1b9c7e3b8d.png)

![image](https://user-images.githubusercontent.com/8952138/115979342-42109000-a53a-11eb-95a7-dbaf0fb3c6af.png)

**Reply by Limezy** [reply]
Oh yeah an internal "time machine".
But why daily pages only ? Should be available for everypage, especially when datahike will come around

**Reply by jefftangx** [reply]
@Limezy Side note, but I believe Daily Pages should have a derived/materialized view, meaning if we edit a page like `Athens` or `Clojure` on the date `March 17, 2021`, but not from the `March 17, 2021` daily notes page, we should still be able to view all edits made on that day.
